adapter/abstract.php

```diff
@@ -50,6 +62,17 @@ public function setPath($path)
 
 		$this->_metadata = null;
 
+        $parts = explode('://', $this->_path);
```

Comment:
@amazeika haven't we decided to drop `://` in favor of parse_url ?

Reply:
@ercanozkaya Yes, it's cleaner. Will make it so.